### PR TITLE
ruff: fix docs in src/

### PR DESCRIPTION
**Pull request by Borda**
# Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure to update the docs?
- [x] Did you write any new necessary tests?

## What does this PR do?

Fixing docstring in source/package

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: src/lightning_utilities/core/rank_zero.py [hunk 1]
```diff
@@ -57,37 +60,44 @@ def _warn(message: Union[str, Warning], stacklevel: int = 2, **kwargs: Any) -> N
...
         # specify the rank of the process being logged
         return f"[rank: {rank}] {message}"
     return message
 
 
 class WarningCache(set):
+    """Cache for warnings."""
+
     def warn(self, message: str, stacklevel: int = 5, **kwargs: Any) -> None:
+        """Proceed warning message."""
```
Comment (carmocca) [inline src/lightning_utilities/core/rank_zero.py:88]: Or emit as above

```suggestion
        """Trigger warning message."""
```

File: src/lightning_utilities/core/rank_zero.py [hunk 2]
```diff
@@ -57,37 +60,44 @@ def _warn(message: Union[str, Warning], stacklevel: int = 2, **kwargs: Any) -> N
...
         if message not in self:
             self.add(message)
             rank_zero_warn(message, stacklevel=stacklevel, **kwargs)
 
     def deprecation(self, message: str, stacklevel: int = 6, **kwargs: Any) -> None:
+        """Proceed deprecation message."""
```
Comment (carmocca) [inline src/lightning_utilities/core/rank_zero.py:94]:
```suggestion
        """Trigger deprecation message."""
```

File: src/lightning_utilities/core/rank_zero.py [hunk 3]
```diff
@@ -57,37 +60,44 @@ def _warn(message: Union[str, Warning], stacklevel: int = 2, **kwargs: Any) -> N
...
         if message not in self:
             self.add(message)
             rank_zero_deprecation(message, stacklevel=stacklevel, **kwargs)
 
     def info(self, message: str, stacklevel: int = 5, **kwargs: Any) -> None:
+        """Proceed info message."""
```
Comment (carmocca) [inline src/lightning_utilities/core/rank_zero.py:100]:
```suggestion
        """Trigger info message."""
```

File: src/lightning_utilities/core/rank_zero.py [hunk 4]
```diff
@@ -57,37 +60,44 @@ def _warn(message: Union[str, Warning], stacklevel: int = 2, **kwargs: Any) -> N
...
 
 
 def rank_zero_deprecation(message: Union[str, Warning], stacklevel: int = 5, **kwargs: Any) -> None:
+    """Raise deprecation warning only for rank zero calls."""
     category = kwargs.pop("category", rank_zero_deprecation_category)
     rank_zero_warn(message, stacklevel=stacklevel, category=category, **kwargs)
 
 
 def rank_prefixed_message(message: str, rank: Optional[int]) -> str:
+    """Wrap a rant type message."""
```
Comment (carmocca) [inline src/lightning_utilities/core/rank_zero.py:77]:
```suggestion
    """Add a prefix with the rank to a message."""
```

File: src/lightning_utilities/core/rank_zero.py [hunk 5]
```diff
@@ -57,37 +60,44 @@ def _warn(message: Union[str, Warning], stacklevel: int = 2, **kwargs: Any) -> N
...
-    """Function used to log warn-level messages only on global rank 0."""
+    """Emit warn-level messages only on global rank 0."""
     _warn(message, stacklevel=stacklevel, **kwargs)
 
 
 rank_zero_deprecation_category = DeprecationWarning
 
 
 def rank_zero_deprecation(message: Union[str, Warning], stacklevel: int = 5, **kwargs: Any) -> None:
+    """Raise deprecation warning only for rank zero calls."""
```
Comment (carmocca) [inline src/lightning_utilities/core/rank_zero.py:71]:
```suggestion
    """Emit a deprecation warning only on global rank 0."""
```

File: src/lightning_utilities/core/rank_zero.py [hunk 6]
```diff
@@ -13,7 +13,10 @@
 
 
 def rank_zero_only(fn: Callable) -> Callable:
-    """Function that can be used as a decorator to enable a function/method being called only on global rank 0."""
+    """Wrap a function to call internal function only in rank zero.
+
+    Function that can be used as a decorator to enable a function/method being called only on global rank 0.
```
Comment (carmocca) [inline src/lightning_utilities/core/rank_zero.py:18]: What was wrong with the previous description?

```suggestion
    """Function that can be used as a decorator to enable a function/method being called only on global rank 0.
```

File: src/lightning_utilities/core/overrides.py [hunk 7]
```diff
@@ -8,6 +8,7 @@
 
 
 def is_overridden(method_name: str, instance: object, parent: Type[object]) -> bool:
+    """Check if method of given object was overwritten."""
```
Comment (carmocca) [inline src/lightning_utilities/core/overrides.py:11]:
```suggestion
    """Check if a method of a given object was overwritten."""
```

File: src/lightning_utilities/core/inheritance.py [hunk 8]
```diff
@@ -15,4 +17,5 @@ def recurse(cl: Type) -> Iterator[Type]:
 
 
 def get_all_subclasses(cls: Type) -> Set[Type]:
+    """Get listed all subclasses."""
```
Comment (carmocca) [inline src/lightning_utilities/core/inheritance.py:20]:
```suggestion
    """List all subclasses of a class."""
```

File: src/lightning_utilities/core/rank_zero.py (hunk 6 above)
Comment (Borda) [reply src/lightning_utilities/core/rank_zero.py:18]: it is not declarative

**Review by akihironitta — APPROVED** [review]

**Pull request merged**